include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -182,14 +194,16 @@ auto from_json_array_impl(const BasicJsonType& j, ConstructibleArrayType& arr, p
 {
     using std::end;
 
-    arr.reserve(j.size());
+    ConstructibleArrayType ret;
+    ret.reserve(j.size());
```

Comment:
As above, forces an allocation unnecessarily.

Reply:
This one is a bit trickier, we did not constraint on `resize` yet, but I can add it anyway.
I guess that custom containers supporting `reserve` but not `resize` are quite unusual.

Reply:
Ah, missed that this was `reserve` and not `resize`.  Thought it was sizing and then writing to the contents.  This one would require a clear or a new and copy, so maybe it's okay as it is in the PR.